workflow/metrics/server.go

```diff
@@ -138,7 +138,11 @@ func (m *Metrics) garbageCollector(ctx context.Context) {
 			return
 		case <-ticker.C:
 			for key, metric := range m.customMetrics {
-				if time.Since(metric.lastUpdated) > m.metricsConfig.TTL {
+				if metric.realtime {
```

Comment:
I'd prefer this as
```
if time.Since(metric.lastUpdated) > m.metricsConfig.TTL {
  switch {
    case metric.metric.IsRealtime() && metric.completed:
        delete(m.customMetrics, key)
    case !metric.metric.IsRealtime():
        delete(m.customMetrics, key)
  }
}
```

Or perhaps read IsRealtime() just once and cache, but test TTL because it's the primary test here. The result is identical, I'm aiming for understandability.

Reply:
I originally considered writing it like you did, but when it comes to realtime metrics that are not yet complete, there's no need to calculate whether they exceed the TTL, which can save some time.
How do you think about it?

Reply:
I'd prefer readability and understandability over performance here. It's not performance critical nor happening very frequently.

Reply:
Got it, done.